dom/src/main/scala/fs2/dom/MutationObserver.scala

```diff
@@ -0,0 +1,50 @@
+package fs2.dom
+
+import cats.syntax.all._
+import org.scalajs.dom
+import cats.effect.kernel.Async
+import cats.effect.std.Dispatcher
+import cats.effect.kernel.Resource
+import cats.effect.kernel.Sync
+
+abstract class MutationObserver[F[_]] private[dom] {
+
+  def observe(target: Node[F], options: dom.MutationObserverInit): F[Unit]
+
+  def disconnect: F[Unit]
+
+  def takeRecords: F[List[dom.MutationRecord]]
```

Comment:
Shoot, I wasn't paying enough attention last time. Even though `dom.MutationRecord` and `dom.ResizeObserverEntry` are immutable objects which we don't need to wrap, they are referencing many things which we probably do want to wrap. Gah.